text/0240-app-telemetry-schema-vision-roadmap.md

```diff
@@ -0,0 +1,305 @@
+ranging from the most general to one that is specifically refined for an
+application. Each child schema inherits the properties of the parent schema and
+can, if necessary, override these properties locally. Additionally, any
+telemetry schema can import one or several semantic convention registries,
+allowing for the definition of OpenTelemetry, vendor-specific, and
+enterprise-level registries. These two enhancements make OpenTelemetry more
+customizable, extensible, and ecosystem-friendly.
+
+The following section will elaborate on the concepts of the Application
+Telemetry Schema, schema inheritance, semantic convention registries, Resolved
+Telemetry Schema, and their relationship with both the existing OpenTelemetry
+schema v1.1 and the OpenTelemetry semantic conventions.
+
+## Proposal
+
+### Overview
+
+Conceptually, this proposal is based on three main concepts: Telemetry
+Schema v1.2, Semantic Convention Registry, and Resolved Telemetry Schema. The
+relationships between these entities are described in the following diagram.
+
+![Application Telemetry Schema Concepts](./img/0240-app-telemetry-schema-concepts.png)
+
+The Telemetry Schemas form a hierarchy of schemas, the complexity of which
+depends on the complexity of the situation (usage in very large enterprises
+will tend to produce more complex schema hierarchies). A single Telemetry Schema
+can import from zero to n Semantic Convention Registries, depending on its
+needs. A Resolved Telemetry Schema is generated during the schema resolution
+process from a hierarchy of Telemetry Schemas and Semantic Convention
+Registries.
+
+The following diagram illustrates a possible instance of a complex hierarchy of
+schemas and semantic convention registries. It involves several vendor and
+enterprise artifacts, in addition to the standardized OpenTelemetry artifacts.
+The schema resolution process will produce a self-contained Resolved Telemetry
+Schema that can be easily consumed by various tools and applications, such as a
+Client SDK generator, compatibility checker, compliance checker, data catalog
+feeder, and more.
+
+![Example of Application Telemetry Schema Hierarchy](./img/0240-app-telemetry-schema-hierarchy-example.png)
+
+For each important component, the following diagram defines the responsibilities
+and key design properties that have been considered in the subsequent sections
+of this document.
+
+![Application Telemetry Schema - Levels of responsibilities & Key design properties](./img/0240-app-telemetry-schema-design-properties.png)
+
+This design enables the definition of semantic conventions in a distributed
+manner. OpenTelemetry, vendors, and enterprises can define their own semantic
+conventions in different registries simplifying the existing process.
+
+The next two sections will describe the concepts of Telemetry Schema v1.2 and
+the Resolved Telemetry Schema.
+
+## Telemetry Schema v1.2 (aka Application Telemetry Schema)
```

Comment:
Currently, the telemetry schemas [v1.1.0 file format](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/schemas/file_format_v1.1.0.md) only defines the changes in the schema from version to version. Presumably, you aim to expand this to be able to declare the schema itself, and not just the changes. If so, big 👍 from me.

Reply:
Yes this is exactly the intent.